lib/src/Pos/Diffusion/Full.hs

```diff
@@ -322,7 +323,8 @@ runDiffusionLayerFull networkConfig ourVerInfo oq slotDuration listeners action
         return ((>>= either throwM return) <$> itMap)
     subscriptionThread nc sactions = case topologySubscriptionWorker (ncTopology nc) of
         Just (SubscriptionWorkerBehindNAT dnsDomains) -> do
-            dnsSubscriptionWorker oq networkConfig dnsDomains slotDuration sactions
+            timer <- newDynamicTimer (convertUnit <$> slotDuration)
+            dnsSubscriptionWorker (OQ.updatePeersBucket oq) networkConfig dnsDomains timer sactions
```

Comment:
Could you put a TODO note here about how we need to wire this timer up somewhere else so that it resets when we receive a block?